pkg/apis/serving/k8s_validation.go

```diff
@@ -609,3 +626,18 @@ func ValidateNamespacedObjectReference(p *corev1.ObjectReference) *apis.FieldErr
 	}
 	return errs
 }
+
+// This is attached to contexts as they are passed down through a container
+// being validated.
+type singleContainer struct{}
+
+// WithinSingleContainer notes on the context that further validation or defaulting
+// is within the context of a single container in the revision.
+func WithSingleContainer(ctx context.Context) context.Context {
+	return context.WithValue(ctx, singleContainer{}, struct{}{})
+}
+
+// Check if we are in the context of a single container in the revision.
+func IsSingleContainer(ctx context.Context) bool {
+	return ctx.Value(singleContainer{}) != nil
+}
```

Comment:
This could be moved elsewhere if there is a place for these helpers related to context.

Reply:
I think we should be able to get rid of them completely.  See my comment above.